src/SocketContext.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import io from 'socket.io-client';
+import { useAuth } from './AuthProvider';
+import NotificationModal from '../src/components/common/notificationModal';
+
+const SocketContext = createContext();
+
+export const SocketProvider = ({ children }) => {
+  const { userData } = useAuth();
+  const [socket, setSocket] = useState(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+
+  useEffect(() => {
+    if (!userData) {
+      console.error('No user data found');
+      return;
+    }
+
+    const userId = userData._id;
+
+    const socketIo = io('http://localhost:8000', {
+      query: { userId: userId },
+    });
+
+    socketIo.on('connect', () => {
+      console.log('Connected with socket ID:', socketIo.id);
+    });
+
+    socketIo.on(`applications-${userId}`, (data) => {
```

Comment:
These three functions are practically identical. The second parameter of `socketIo` is basically a `handleNotification` function. While the first parameter is the only dynamic value. That'll clean things up a lot!

Reply:
Thank you!
Done✅
Added setupSocketListeners function to manage socket events